service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/CodeableConceptCdMapper.java

```diff
@@ -77,12 +77,16 @@ public String mapCodeableConceptToCdForTransformedActualProblemHeader(CodeableCo
                 return Optional.of(codeableConcept.getText());
             } else {
                 var extension = retrieveDisplayExtension(coding.get());
-                return extension.map(value -> value
-                    .getExtension()
-                    .stream()
+                Optional<String> originalText = extension.stream()
                     .filter(extension1 -> extension1.getUrl().equals(DESCRIPTION_DISPLAY))
```

Comment:
Not sure if this shouldn't be following the principle of checking if CONST equals something in order to handle the NullpointerException case?
```suggestion
                    .filter(extension1 -> DESCRIPTION_DISPLAY.equals(extension1.getUrl()))
```